### ISSUE TITLE
Should we create a git tag for ourselves?

### ISSUE THREAD

**Issue by sgalcheung** [florian-lefebvre/astro-integration-template#142]
I didn't see any git tag command in [release file](https://github.com/florian-lefebvre/astro-integration-template/blob/main/scripts/release.mjs), but I found some tags in your [astro-integration-kit](https://github.com/florian-lefebvre/astro-integration-kit/tree/main).

So, Should we create a git tag for ourselves?

**Comment by florian-lefebvre** [comment]
I think [this step](https://github.com/florian-lefebvre/astro-integration-template/blob/main/scripts/release.mjs#L44) is responsible for tags, so it should work out of the box when using this template already. Are you facing some issues?

**Comment by sgalcheung** [comment]
You mentioned that [the step](https://github.com/florian-lefebvre/astro-integration-template/blob/134fdc283b27071a1257229a8e295640fa1dc1eb/scripts/release.mjs#L44) is a push tag following the git push commit, but we haven't created a tag for a push.

**Comment by florian-lefebvre** [comment]
I think changeset handles it. FYI https://github.com/florian-lefebvre/astro-loading-indicator is based on this template, I've not changed the release file and still have some github tags so I assume it's working

**Comment by sgalcheung** [comment]
> [Publishes to NPM repo, and creates git tags. ](https://github.com/changesets/changesets/blob/f03f70bffc9a033747f45b0d071c6c2ed8818ac7/packages/cli/README.md?plain=1#L161)

You remind me, `changeset publish [--otp={token}]` will indeed handle it, I think I could meet a 2FA problem, this resulted in no git tag being created.

**Comment by florian-lefebvre** [comment]
Ah that's a good point! Do you want to submit a PR with a comment in the release script?

**Comment by florian-lefebvre** [comment]
I was thinking specifically about the following
```diff
+ // You may need to use the --otp flag for 2FA, see https://github.com/changesets/changesets/blob/f03f70bffc9a033747f45b0d071c6c2ed8818ac7/packages/cli/README.md#publish
await run("pnpm changeset publish");
```

**Comment by sgalcheung** [comment]
But this token is not should in public.

**Comment by florian-lefebvre** [comment]
could that work using env vars? For example `pnpm changeset publish --otp OTP`, and then run `OTP=foo pnpm release`

**Comment by sgalcheung** [comment]
I use a method to handle env vars:

```
function parseArgs(argv) {
  const params = {};

  for (let i = 0; i < argv.length; i++) {
    if (argv[i].startsWith("--")) {
      const [key, value] = argv[i].includes("=")
        ? argv[i].slice(2).split("=")
        : [
            argv[i].slice(2),
            argv[i + 1] && !argv[i + 1].startsWith("--") ? argv[++i] : true,
          ];
      params[key] = value;
    } else if (argv[i].includes("=")) {
      const [key, value] = argv[i].split("=");
      params[key] = value;
    }
  }

  return { ...process.env, ...params };
}

if (params.OTP) {
    await run(`pnpm changeset publish --otp=${params.OTP}`);
  } else {
    await run("pnpm changeset publish");
  }


```

Finally, I tested it, and it seems to work well.

> `>` changeset "publish" "--otp=362675"
> 
> 🦋  info npm info desquidex
> 🦋  info desquidex is being published because our local version (0.1.1) has not been published on npm
> 🦋  info Publishing "desquidex" at "0.1.1"
> 🦋  success packages published successfully:
> 🦋  desquidex@0.1.1
> 🦋  Creating git tag...
> 🦋  New tag:  desquidex@0.1.1

**Comment by florian-lefebvre** [comment]
I think we can use https://nodejs.org/api/util.html#utilparseargsconfig now. Anyways, I'd love a PR for this, it is useful!
